backend/server/services/userService.js

```diff
@@ -46,36 +46,72 @@ const createUser = async (email, password, confirmPassword) => {
 
 const loginUser = async (email, password) => {
   try {
-    var noSuchUser = Boolean();
+    var userId = Number();
 
     // Check for missing inputs
     if (!email || !password) {
       throw { status: 400, message: 'Missing inputs' };
     }
 
     // Check if a user with the given email exists
-    const existingUserCheck = userDatabase
-      .findByEmail(email)
-      .then(userId => { 
-        noSuchUser = userId == null; 
-      });
-    if (noSuchUser) {
-      throw { status: 400, message: 'User not found' };
+    await userDatabase.findByEmail(email)
+      .then(id => userId = id);
+  
+    if (!userId) {
+      throw { status: 400, message: 'Email not registered with any user' };
     }
 
     // Compare the entered password with the hashed password stored in the database
-    const passwordMatch = await bcrypt.compare(password, user.password);
-    if (!passwordMatch) {
+    // FIXME: need to find a way to retrieve stored hashed password
+    if (!bcrypt.compareSync(password, user.password)) {
       throw { status: 400, message: 'Incorrect password' };
     }
 
-    return user;
+    return userId;
+  } catch (err) {
+    throw err;
+  }
+};
+
+const getUserInfo = async (userId, email) => {
+  try {
+    if (!userId && !email) {
+      throw { status: 400, message: 'Need at least id or email' };
+    }
+
+    if (userId != null)
+         return userDatabase.getUserInfoById(userId);
```

Comment:
I think this is overindented

Reply:
It is, but its to match with the line below.
![image](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g32/assets/88828097/96ef5dc2-26af-4893-9990-476cf51cdd2c)